### ISSUE TITLE
Improvements for tab titles

### ISSUE THREAD

**Issue by Foalyy** [sublimehq/sublime_merge#1072]
**Problem description**

I use Sublime Merge for tracking the code of the firmwares of my embedded projects. Since this code is only one part of the project, I tend to have it in a sub-directory like this : `myprojectname/firmware`, alongside other documents. Since Sublime Merge uses the directory name as the tab title, I end up with multiple tabs simply named `firmware`, which are hard to track.
This seems specific, but I wouldn't be surprised if other users have the same "issue" (admittedly small one) with different workflows.

**Preferred solution**

It would be useful to be able to set an alias for each tab (through a right-click menu for instance). I understand that the philosophy (which I like) of Sublime Merge is to be a wrapper around git that doesn't its own config files in the repository where such alias could be stored, therefore it could simply be kept in the GUI state as long as the tab is open, and forgotten when it's closed.
In a more general (and long-term) way, I think it would be nice to have more information in the tabs : for example a color or an icon to indicate whether the working tree is clean, the name of the current branch, etc.

**Comment by poldi1405** [comment]
I think a right-click menu would not be the greatest idea since it would then have exactly one entry which I think would not make much sense. Relating to my proposed solution this might be possible by setting a custom git-config value instead? Not sure on that one though.

**Comment by michaelalandawson** [comment]
I also have a couple open repos that have the same directory name.  I would like to be able to rename the tab or have an alias per repo.

**Comment by RamiroMaldo** [comment]
I find this feature really useful too. I've developed a framework and each application based on it is built in a framework/app sub-directory, so I can have the main framework repo and the app repos separately. Just as @Foalyy mention, when I have more than one app repo open on SM (ie. all the time TBH), I cannot tell easily which one is.
GitHub Desktop has this repo "alias" workaround implemented, but I prefer SM a million times! :)
Please consider adding this feature. Thanks!

**Comment by alanfluff** [comment]
+1

It is so easy to get lost when repo Tab name just === directory name. Naming repos seems such a fundamental ability I can't quite believe it is still not possible with SM.

I would pay for an official add-on, *just* to be able to do this, it is such a gaping hole in an otherwise superb app.

Pretty please with sprinkles? ( thank you : )

**Comment by mpldr** [comment]
I think an easy solution to this would be to learn from good old cgit. `git config sublime-merge.label "New Name to show in tab"` and SM just uses the user provided name.

**Comment by alanfluff** [comment]
> I think an easy solution to this would be to learn from good old cgit. `git config sublime-merge.label "New Name to show in tab"` and SM just uses the user provided name.

wow!? Gonna try that -- if it works how it sounds, then *thank you* @mpldr

**Comment by mpldr** [comment]
It does not (at least not yet), it was just a suggestion to the devs at sublime at how to have this handled in an incredibly easy way.

**Comment by alanfluff** [comment]
Yes. That would be an excellent solution. Weird this has languished so long.

**Comment by mbartlett21** [comment]
Perhaps this could be handled like how Sublime Text does with multiple tabs open with the same filename, but different directories:

![image](https://user-images.githubusercontent.com/29034492/233289332-aa4022ab-410d-4c75-b485-199587023748.png)

(showing the portion of the full path that differs)

**Comment by pacanukeha** [comment]
I would love this too.  All of my different projects have git files in a src directory underneath other stuff (IDE settings, non-source-code-controlled documentation, other project management stuff).  I could force myself to rename everything as a hack but it seems strange that this feature is missing and has been languishing for so long.

**Comment by dpjohnst** [comment]
Hi all,

We've added support for tab aliases in Sublime Merge 2086 (development build) :tada:
This will be available in the next stable build too.

Kind regards,
\- Dylan from Sublime HQ

**Comment by willrowe** [comment]
Would it be possible to use the alias in the window name and repository switcher as well?

**Comment by themilkman** [comment]
+1 for the suggestion of @willrowe, I had the same idea shortly after the first usage of the new feature. That'd make the aliasing feature much more useful.

**Comment by Foalyy** [comment]
Just tried it, it works great, thanks @dpjohnst ! For me it's fine as it is, but I leave the issue open for now for @willrowe 's suggestion.

**Comment by themilkman** [comment]
Maybe we should open a new issue with two additional feature requests from above as yours is fulfilled :-)

**Comment by themilkman** [comment]
@willrowe : Opened new issue with your suggestion #1799
